code/datums/diseases/advance/symptoms/vision.dm

```diff
@@ -75,3 +75,94 @@ Bonus
 						eyes.forceMove(get_turf(M))
 				else
 					to_chat(M, "<span class='userdanger'>Your eyes burn horrifically!</span>")
+
+/datum/symptom/ocularsensitivity
+	name = "Ocular Hyper-reception"
+	desc = "The virus increases the sensitivity of the rods and cones to light, allowing better vision in the dark"
+	stealth = 1
+	resistance = -2
+	stage_speed = -2
+	transmittable = 0
+	level = 9
+	severity = 0
+	base_message_chance = 50
+	symptom_delay_min = 15
+	symptom_delay_max = 50
+	thresholds = list()//An Associative List of Threshhold related vars. ["better_vis"], ["nvgs"], ["thermal"], ["xray"], ["nvision_buff"], ["other_buff"] are the indexes
+	threshold_desc = "<b>Resistance 10:</b> Vision in the dark is even better.<br>\
+					  <b>Resistance 14:</b> The host will grow a pair of NVGs.<br>\
+					  <b>Stage Speed 10:</b> The host's eyes are able to percieve infrared radiation.<br>\
+					  <b>Stage Speed 18:</b> The host's eyes are able to percieve xray radiation. This overrides the thermal vision threshhold."
+
+/datum/symptom/ocularsensitivity/severityset(datum/disease/advance/A)
+	. = ..()
+	if(A.properties["resistance"] >= 10) //Better Night Vision
+		severity -= 1
+	if(A.properties["resistance"] >= 14)
+		severity += 2 //Disregards the decrease from other resistance threshold, in fact is probably borderline antagonistic.
+	if(A.properties["stage_rate"] >= 10) //Thermal Vision
+		severity -= 1
+	if(A.properties["stage_rate"] >= 18) //Xray Vision
+		severity -= 2
+
+/datum/symptom/ocularsensitivity/Start(datum/disease/advance/A)
+	. = ..()
+	if(A.properties["resistance"] >= 10) //Better Night Vision
+		thresholds["better_vis"] = TRUE
+	if(A.properties["resistance"] >= 14) //Grow a pair of NVGS
+		thresholds["nvgs"] = TRUE
+	if(A.properties["stage_rate"] >= 10) //Thermal Vision
+		thresholds["thermal"] = TRUE
+	if(A.properties["stage_rate"] >= 18) //Xray Vision
+		thresholds["xray"] = TRUE
+
+/datum/symptom/ocularsensitivity/Activate(datum/disease/advance/A)
+	if(!..())
+		return
+	var/mob/living/L = A.affected_mob
+	switch(A.stage)
+		if(4)
+			if(!thresholds["nvision_buff"])
+				to_chat(L, "<span class='userdanger'>Your vision becomes better!</span>")
+				if(thresholds["better_vis"])
+					L.see_in_dark += 4
+					thresholds["nvision_buff"] = TRUE
+				else
+					L.see_in_dark += 2
+					thresholds["nvision_buff"] = TRUE
+			if(thresholds["thermal"] && !thresholds["xray"] && !thresholds["other_buff"])
+				ADD_TRAIT(L, TRAIT_THERMAL_VISION, DISEASE_TRAIT)
+				thresholds["other_buff"] = TRUE
+			if(thresholds["xray"] && thresholds["other_buff"])
+				ADD_TRAIT(L, TRAIT_XRAY_VISION, DISEASE_TRAIT)
+				thresholds["other_buff"] = TRUE
```

Comment:
```suggestion
			if(!thresholds["other_buff"])
				thresholds["other_buff"] = TRUE
				if(thresholds["xray"])
					ADD_TRAIT(L, TRAIT_XRAY_VISION, DISEASE_TRAIT)
				else if(thresholds["thermal"])
					ADD_TRAIT(L, TRAIT_THERMAL_VISION, DISEASE_TRAIT)
```

If you want a cleaner code, you should merge `thresholds["other_buff"]` and `thresholds["nvision_buff"]` into a single var which serves as an indicator whether it has given the affecting mob corresponding buffs.